### PR TITLE
Add dilation of the segmentation mask to improve registration robustness

### PR DESCRIPTION
**Pull request by PeterBehringer**
This pull request fixes issue #24

### CODE REVIEW

File: SliceTracker.py [hunk 1]
```diff
@@ -2543,6 +2571,24 @@ def applyReRegistration(self,fixedVolume,movingVolume,fixedLabel,targets,lastRig
...
+
+      import SimpleITK as sitk
+      import sitkUtils
+
+      print 'mask '+str(mask.GetName()+' is dilated')
+
+      labelImage = sitk.ReadImage(sitkUtils.GetSlicerITKReadWriteAddress(mask.GetName()))
+
+      grayscale_dilate_filter = sitk.GrayscaleDilateImageFilter()
+      grayscale_dilate_filter.SetKernelRadius([12,12,0])
```
Comment (fedorov) [inline SliceTracker.py]: Peter, how did you come up with 12?

File: SliceTracker.py [hunk 2]
```diff
@@ -2543,6 +2571,24 @@ def applyReRegistration(self,fixedVolume,movingVolume,fixedLabel,targets,lastRig
...
+      grayscale_dilate_filter.SetKernelType(sitk.sitkBall)
+      labelImage=grayscale_dilate_filter.Execute(labelImage)
+
+      sitk.WriteImage(labelImage, sitkUtils.GetSlicerITKReadWriteAddress(mask.GetName()))
+      print 'dilate mask through'
```
Comment (fedorov) [inline SliceTracker.py]: should be removed or use logging instead

**Comment by fedorov** [comment]
While looking for dilate, I came across this line: https://github.com/SlicerProstate/SliceTracker/blob/reg_update/SliceTracker.py#L1843. Can you explain what is the purpose of that?

**Comment by fedorov** [comment]
Looking at the applyBSplineRegistration() - I don't see dilateMask() applied. Am I missing something?

https://github.com/SlicerProstate/SliceTracker/blob/reg_update/SliceTracker.py#L2128

File: SliceTracker.py (hunk 2 above)
Comment (PeterBehringer) [inline SliceTracker.py]: ok

File: SliceTracker.py (hunk 1 above)
Comment (PeterBehringer) [inline SliceTracker.py]: I tried out different sizes at project week. 12 was the best for most cases.

**Comment by PeterBehringer** [comment]
Yes, that is correct. At project week, we investigated improvement of robustness of registration only for intraop to intraop, so I did not add it to preop to intraop registration without discuss this with you. Ill add and commit it.

**Comment by PeterBehringer** [comment]
My idea was, that the user only need to segment every second slice, and click on the 'apply' afterwards. Then, an automatic dilation follows in order to fill the slices in between.

https://github.com/SlicerProstate/SliceTracker/blob/reg_update/SliceTracker.py#L1843

**Comment by fedorov** [comment]
I see. I was confused by the name of the function. It is called 'erode', but in fact it is a dilation operation implemented as erode of the background value.

**Comment by fedorov** [comment]
About adding dilate to preop-intraop - this should be tested first to confirm that it works as expected. Another idea is to add this as an advanced option flag that can be enabled by the user if the default preop-intraop registration does not work well.

**Comment by che85** [comment]
@PeterBehringer Does adding dilate to preop-intraop work as expected? I think you are the one who can test that properly since you wrote that code.

I would like to merge the pull request as soon as possible to continue with refactoring of the registrations methods...

**Comment by fedorov** [comment]
Sorry for the delay! We can test and add dilation for preop later, this is not urgent, since the results seem to be good.

**Pull request merged**